// Copyright 2023 Ant Group Co., Ltd.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//   http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//

syntax = "proto3";

import "secretflow_serving/apis/common.proto";
import "secretflow_serving/apis/status.proto";
import "secretflow_serving/protos/bundle.proto";

package secretflow.serving.apis;

option cc_generic_services = true;

// ModelService provides operation ralated to models.
service ModelService {
  rpc GetModelInfo(GetModelInfoRequest) returns (GetModelInfoResponse);
}

message GetModelInfoRequest {
  // Custom data.
  Header header = 1;

  // Model service specification.
  ServiceSpec service_spec = 2;
}

message GetModelInfoResponse {
  // Custom data.
  Header header = 1;

  // Staus of this response.
  Status status = 2;

  // Model service specification.
  ServiceSpec service_spec = 3;

  ModelInfo model_info = 4;
}
